### PR TITLE
fix(credential): audit changes credential

### PR DESCRIPTION
**Pull request by dixitaniket**

### CODE REVIEW

File: x/credential/credential_test.go [hunk 1]
```diff
@@ -552,6 +553,43 @@ func TestWrappedCredential_Validate(t *testing.T) {
...
+				assert.NoError(t, err)
+				_, pubKey, err := kr.SignByAddress(ki.GetAddress(), data)
+				assert.NoError(t, err)
+
+				// attach the proof
+				date := time.Date(2022, 02, 24, 0, 0, 0, 0, time.UTC)
+				wc.Proof = NewProof(
+					pubKey.Type(),
+					date.Format(time.RFC3339),
+					// TODO: define proof purposes
```
Comment (noandrea) [inline x/credential/credential_test.go]: assertionMethod is a good choice for the purpose, why the TODO?

File: x/credential/keeper/genesis.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/elesto-dao/elesto/v2/x/credential"
+)
+
+func InitGenesis(ctx sdk.Context, k Keeper, data *credential.GenesisState) {
+	//TODO: validate data in genesis
```
Comment (noandrea) [inline x/credential/keeper/genesis.go]: What needs to be done here?

**Review by noandrea — COMMENTED** [review]
looks great! let's take care of those 2 TODO then let's smash the big green button !

File: x/credential/keeper/genesis.go (hunk 2 above)
Comment (dixitaniket) [reply x/credential/keeper/genesis.go]: was thinking about some sort of validation for the fields in the genesis state.

**Review by noandrea — APPROVED** [review]
:zap:

**Pull request merged**